gravitee-apim-console-webui/src/management/api/general/details/api-general-info-duplicate-dialog/api-general-info-duplicate-dialog.component.ts

```diff
@@ -113,3 +134,14 @@ const extractContextPath = (api: Api) => {
   }
   return '';
 };
+
+const extractHost = (api: Api) => {
+  if (api.definitionVersion === 'V4') {
+    const apiV4 = api as ApiV4;
+    const tcpListener = apiV4.listeners?.find((listener) => listener.type === 'TCP');
```

Comment:
If you force the type here I think you'll be able to remove the cast below

```suggestion
    const tcpListener: TcpListener = apiV4.listeners?.find((listener) => listener.type === 'TCP');
```